### PR TITLE
create build when none is found

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/paas 

only look at second commit, first is from other PR

### Risks
- Low

### CODE REVIEW

File: plugins/kubernetes/app/models/kubernetes/deploy_executor.rb [hunk 1]
```diff
@@ -79,13 +79,43 @@ def pod_is_live?(release_doc)
...
+        creator: @job.user,
+        project: @job.project,
+        label: "Automated build triggered via Deploy ##{@job.deploy.id}"
+      )
+      DockerBuilderService.new(build).run!(push: true)
+
+      loop do
+        @output.puts("Waiting for Build ##{build.id} to finish.")
+        sleep 2
+        break if build.docker_build_job.finished?
```
Comment (jonmoter) [inline plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: Have a way to break out of this if the user hits the "Cancel" button.

File: plugins/kubernetes/app/models/kubernetes/deploy_executor.rb [hunk 2]
```diff
@@ -79,13 +79,43 @@ def pod_is_live?(release_doc)
...
+      build = Build.create!(
+        git_ref: @job.commit,
+        creator: @job.user,
+        project: @job.project,
+        label: "Automated build triggered via Deploy ##{@job.deploy.id}"
+      )
+      DockerBuilderService.new(build).run!(push: true)
+
+      loop do
+        @output.puts("Waiting for Build ##{build.id} to finish.")
```
Comment (jonmoter) [inline plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: Show URL so end user can go to that page.

File: plugins/kubernetes/app/models/kubernetes/deploy_executor.rb [hunk 3]
```diff
@@ -79,13 +79,43 @@ def pod_is_live?(release_doc)
     # TODO: create it and wait ... also show creation logs etc
     def find_build
       if build = Build.find_by_git_sha(@job.commit)
-        @output.puts("Found build #{build.id} for sha #{@job.commit}")
+        ensure_build_is_successful(build)
       else
-        raise Samson::Hooks::UserError, "Build for sha #{@job.commit} does not exist, create it before deploying"
+        build = create_build
+        ensure_build_is_successful(build)
```
Comment (jonmoter) [inline plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: DRY up by putting outside the "else"

**Comment by jonmoter** [comment]
👍  other than small comments above

**Comment by grosser** [comment]
updated to also wait when build is already running + stop when stopped

**Comment by jonmoter** [comment]
👍

**Pull request merged**